docs/proposals/job.md

```diff
@@ -0,0 +1,135 @@
+# Job Controller
+
+## Abstract
+A proposal for implementing a new controller - Job controller, which will be responsible
+for managing pod(s) that require to run on demand.
+
+Several existing issues and PRs were already created regarding that particular subject:
+* Job Controller [#1624](https://github.com/GoogleCloudPlatform/kubernetes/issues/1624)
+* New Job resource [#7380](https://github.com/GoogleCloudPlatform/kubernetes/pull/7380)
+
+
+## Use Cases
+1. Be able to start of one or several pods tracked as a single entity.
+1. Be able to implement heavy-computation model, eg. MapReduce.
```

Comment:
We need to distinguish workloads that run for a bounded duration and complete on their own from those that perform continuous data processing indefinitely. I generally call the former batch, but they could also be called workflow. Job is a good name for the API resource. I don't want any more resources with controller in the name.

In addition to the features listed, users also want parallelism control (max and min simultaneous instances), task assignment, and queuing. However, I'd start simple.